src/main/java/bio/terra/service/job/JobService.java

```diff
@@ -373,20 +374,26 @@ public Stairway getStairway() {
       throw new InvalidResultStateException("No result map returned from flight");
     }
 
-    switch (flightState.getFlightStatus()) {
-      case FATAL:
-      case ERROR:
+    JobModel.JobStatusEnum jobStatus = getJobStatus(flightState);
+
+    switch (jobStatus) {
+      case FAILED:
+        final Exception exceptionToThrow;
+
         if (flightState.getException().isPresent()) {
-          Exception exception = flightState.getException().get();
-          if (exception instanceof RuntimeException) {
-            throw (RuntimeException) exception;
-          } else {
-            throw new JobResponseException("wrap non-runtime exception", exception);
-          }
+          exceptionToThrow = flightState.getException().get();
+        } else if (getCompletionToFailureException(flightState).isPresent()) {
+          exceptionToThrow = getCompletionToFailureException(flightState).get();
+        } else {
+          exceptionToThrow =
+              new InvalidResultStateException("Failed operation with no exception reported");
```

Comment:
Not saying this is wrong, I'm just curious where this could happen?

Reply:
I think it would be from an internal Stairway error where the flight state doesn't have a resultMap or exception associated with it.